lang/cs/Org.Apache.REEF.DistributedR/TaskRunner.cs

```diff
@@ -63,18 +65,24 @@ public void Start()
             this.toDriverThread.Start();
         }
 
-        public void Stop(bool wait = false)
+        internal void Stop()
         {
             Logr.Log(Level.Info, "Shutting down");
 
-            if (!wait)
+            Submit(ShutdownMsg.Factory());
+            this.cancelToken.WaitHandle.WaitOne(3000);
```

Comment:
This seems like there could be a timing issue here -- is this where you mentioned a long job could be an issue?